eventhandler.go

```diff
@@ -16,33 +16,67 @@ type EventHandler interface {
 	Topic() string
 }
 
-// SetupEventHandler sets up an Eventhandler to the event bus.
-func (b *EventBus) SetupEventHandler(ctx context.Context, eventHandler EventHandler) error {
-	const errMessage = "failed to setup event handler: %w"
+// SetupEventHandlers sets up the given event handlers.
+func (b *EventBus) SetupEventHandlers(ctx context.Context, handlers ...EventHandler) error {
+	const errMessage = "failed to add event handlers: %w"
 
-	if err := b.AddHandlerWithOptions(
-		ctx,
-		eventHandler.Events(),
-		eventHandler,
-		WithHandlerTopic(eventHandler.Topic()),
-	); err != nil {
-		return fmt.Errorf(errMessage, err)
+	for i := range handlers {
+		handler := handlers[i]
+
+		err := b.setupEventHandler(ctx, handler.Events(), handler, handler.Topic())
+		if err != nil {
+			return fmt.Errorf(errMessage, err)
+		}
 	}
 
 	return nil
 }
 
-// SetupEventHandlers sets up the given event handlers.
-func (b *EventBus) SetupEventHandlers(ctx context.Context, handlers ...EventHandler) error {
+// SetupEventHandlersWithMiddleware sets up every given handler with the given middlewares.
+func (b *EventBus) SetupEventHandlersWithMiddleware(
```

Comment:
besser: SetupEventHandlersWithMiddlewares (plural)

Reply:
richtig, in eventhorizon wird halt immer der singular genutzt, kp warum, wollte da in dem vokabular bleiben, finde plural aber auch eig besser

Reply:
dann lass uns doch einfach plural nehmen, wenn wir das beide besser finden